src/services/navigationBar.ts

```diff
@@ -192,6 +193,11 @@ namespace ts.NavigationBar {
                             addTopLevelNodes((<Block>functionDeclaration.body).statements, topLevelNodes);
                         }
                         break;
+
+                    default:
+                        const childrens: Node[] = [];
+                        forEachChild(node, child => { childrens.push(child) });

```

Comment:
Currently it sorts *within* a single level, so e.g. methods of a class will be sorted just among each other. For levels of expressions we probably don't want that (`foo(foo(class Y {}, class X {}), foo(class B {}, class A{}))` should show as `A B X Y`).